HIP/hip-1068.md

```diff
@@ -0,0 +1,528 @@
+    }
+```
+
+`IHRC1068` token methods will be callable via the HTS system contract at `0x167`.
+| Hash  | Selector  | HAPI Service (system contract address)  |
+| ----- | --------- | ------------- |
+| 0x    | approve(address owner, address token, address spender, uint256 amount, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS (0x167)  |
+| 0x    | approveNFT(address owner, address token, address approved, uint256 serialNumber, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS (0x167)  |
+| 0x    | setApprovalForAll(address owner, address token, address operator, bool approved, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS (0x167)  |
+
+
+`IHRC1068` account methods will be callable via the HAS system contract at `0x16a`.
+| Hash  | Selector  | HAPI Service (system contract address)  |
+| ----- | --------- | ------------- |
+| 0x    | hbarAllowance(address spender, AllowanceScope allowanceScope) external returns (int64 responseCode, int256 amount) | CryptoService (0x16a)  |
+| 0x    | hbarApprove(address owner, address spender, int256 amount, AllowanceScope allowanceScope) external returns (int64 responseCode) | CryptoService (0x16a)  |
+
+
+The following `IHRC1068` methods will be directly callable by EOAs from a DApp and do not require a deployed smart contract.
+The contract address in this case is the applicable account or token evmAddress
+| Hash  | Selector  | HAPI Service  | Contract facade address   |
+| ----- | --------- | ------------- | ------------------------- |
+| 0x    | approve(address spender, uint256 amount, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS  | Token address   |
+| 0x    | approveNFT(address approved, uint256 serialNumber, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS  | Token address   |
+| 0x    | setApprovalForAll(address operator, bool approved, AllowanceScope allowanceScope) external returns (int64 responseCode) | HTS  | Token address   |
+| 0x    | hbarAllowance(address spender, AllowanceScope allowanceScope) external returns (int64 responseCode, int256 amount) | CryptoService  | Account address   |
+| 0x    | hbarApprove(address spender, int256 amount, AllowanceScope allowanceScope) external returns (int64 responseCode) | CryptoService  | Account address   |
+
+
+### SDK API
+
+With the introduction of new allowance scopes, the SDK will need to expose the functionality to developers.
+The following transactions will need to be expanded with the scope concept
+
+- AccountAllowanceApproveTransaction
+- AccountAllowanceAdjustTransaction
+- AccountAllowanceDeletionTransaction
+
+### Mirror Node API
+
+The Mirror Node allowance REST APIs would need to be updated to support the applicable scope concept values.
+
+#### getCryptoAllowances
+The `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/crypto` would need to be updated to to support all scopes.
```

Comment:
Do I call this endpoint when want to get information about an allowance for calling smart contracts for example? If that is correct I dislike the "/crypto" at the end. Since we have the scope I would assume "/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances" as endpoint.

Reply:
This is an existing endpoint that lists all the allowances with relation to hbar crypto.
This is distinguished from the `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/nfts` and `/api/v1/accounts/{idOrAliasOrEvmAddress}/allowances/tokens` endpoints. 
See https://testnet.mirrornode.hedera.com/api/v1/docs/